File: solutions/JavaScript/subarray_sum_equals_k_brute_force.js
function subarraySum(nums, k) {
    let result = 0;
    for (let i = 0; i < nums.length; i++) {
        let total = 0;
        for (let j = i; j < nums.length; j++) {
            total += nums[j];
            if (total === k) result++;
        }
    }
    return result;
}

File: solutions/C++/subarray_sum_equals_k_brute_force.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int result = 0;
        for (int i = 0; i < (int)nums.size(); i++) {
            int total = 0;
            for (int j = i; j < (int)nums.size(); j++) {
                total += nums[j];
                if (total == k) result++;
            }
        }
        return result;
    }
};
